addons/mail/static/src/discuss/call/common/rtc_session_model.js

```diff
@@ -1,16 +1,47 @@
 import { Record } from "@mail/core/common/record";
+import { Deferred } from "@web/core/utils/concurrency";
+
+const ASYNC_GETTER_TIMEOUT = 120_000;
 
 export class RtcSession extends Record {
     static id = "id";
     /** @type {Object.<number, import("models").RtcSession>} */
     static records = {};
+    static awaitedRecords = new Map();
     /** @returns {import("models").RtcSession} */
     static get(data) {
         return super.get(data);
     }
+    /** @returns {Promies<import("models").RtcSession>} */
+    static async getWhenReady(id) {
+        let session = this.get(id);
+        if (!session) {
+            let deferred = this.awaitedRecords.get(id);
+            if (!deferred) {
+                deferred = new Deferred();
+                this.awaitedRecords.set(id, deferred);
+                setTimeout(() => {
+                    deferred.reject?.();
+                    this.awaitedRecords.delete(id);
+                }, ASYNC_GETTER_TIMEOUT);
+            }
+            try {
+                session = await deferred;
+            } catch {
+                return;
+            }
```

Comment:
There is no point rejecting to immediately catching and returning undefined, it could immediately resolve undefined in the timeout and return the promise in top level.